staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -148,3 +156,33 @@ type CustomResourceValidation struct {
 	// OpenAPIV3Schema is the OpenAPI v3 schema to be validated against.
 	OpenAPIV3Schema *JSONSchemaProps `json:"openAPIV3Schema,omitempty" protobuf:"bytes,1,opt,name=openAPIV3Schema"`
 }
+
+// CustomResourceSubResources defines the status and scale subresources for CustomResources.
+type CustomResourceSubResources struct {
+	// Status denotes the status subresource for CustomResources
+	Status *CustomResourceSubResourceStatus `json:"status,omitempty" protobuf:"bytes,1,opt,name=status"`
+	// Scale denotes the scale subresource for CustomResources
+	Scale *CustomResourceSubResourceScale `json:"scale,omitempty" protobuf:"bytes,2,opt,name=scale"`
+}
+
+// CustomResourceSubResourceStatus defines how to serve the HTTP path <CR Name>/status.
+// status is represented by the `.status` JSON path.
+type CustomResourceSubResourceStatus struct{}
+
+// CustomResourceSubResourceScale defines how to serve the HTTP path <CR name>/scale.
+type CustomResourceSubResourceScale struct {
+	// required, e.g. “.spec.replicas”.
+	// Only JSON paths without the array notation are allowed.
+	SpecReplicasPath string `json:"specReplicasPath" protobuf:"bytes,1,opt,name=specReplicasPath"`
```

Comment:
not opt

Reply:
This should be fixed when proto generation is fixed. Currently, `update-generated-protobuf` fails on `CustomResourceSubResourceStatus`